app/src/components/GameRoutes.tsx

```diff
@@ -98,7 +103,7 @@ function CurrentPlayerProvider(props: {
       }
     },
     onError: _ => {
-      setRedirectRoute(routes.root)
+      currentAuth.setJwtToken(null)
```

Comment:
you can error here, if you are already in a game, but get a new link.

```
graphql-engine_1  | {"type":"http-log","timestamp":"2020-05-15T00:22:45.960+0000","level":"error","detail":{"operation":{"user_vars":{"x-hasura-role":"anonymous"},"error":{"path":"$.selectionSet.players.selectionSet.username","error":"field \"username\" not found in type: 'players'","code":"validation-failed"},"request_id":"663ceaae-57ad-4a40-a4df-72e8e79f3342","response_size":140,"query":{"variables":{"clientUuid":"e7e299ff-0e76-494f-875c-16ce55caefb8","joinCode":"ZOWY"},"operationName":"CurrentPlayer","query":"query CurrentPlayer($clientUuid: uuid!, $joinCode: String!) {\n  players(where: {client_uuid: {_eq: $clientUuid}, game: {join_code: {_eq: $joinCode}}}) {\n    id\n    client_uuid\n    username\n    game {\n      id\n      host {\n        id\n        username\n        __typename\n      }\n      __typename\n    }\n    __typename\n  }\n}\n"}},"http_info":{"status":200,"http_version":"HTTP/1.1","url":"/v1/graphql","ip":"172.20.0.1","method":"POST","content_encoding":null}}}
```

the real solution here is to actually check what game you are signing in for, and if the join code doesn't match you should be signing up instead of signing in anyways.